src/category_theory/quotient.lean

```diff
@@ -0,0 +1,101 @@
+import category_theory.functor
+import category_theory.eq_to_hom
+
+/-
+  This file defines quotient categories. Given a category and an arbitrary family of relations
+  on its homsets, thought of as identifying some homs, we construct a new category on an
+  identical type of objects, where related homs are identified.
+
+    Note that we do *not* assume that our relation behaves well under composition, or that it is
+  an equivalence relation.
+-/
+
+universes v v₁ u u₁
+
+namespace category_theory
+
+variables
+  {C : Type u} [category.{v} C]
+  (r : Π (a b : C), (a ⟶ b) → (a ⟶ b) → Prop)
+include r
+
+/-- The type of objects in the quotient category --/
+structure quotient := (to_C : C)
```

Comment:
I don't like `to_C`... I'll try to think how to name things here, but feel free to proceed if you have an idea.

Reply:
The only reason I don't like `to_C` is that the category may not be called `C` at the point of use.

Reply:
Any of `to`, `as`, or `of`?

Reply:
Yup, I would also like one of those (or maybe even `mk`, but that could be considered abuse of terminology :stuck_out_tongue_winking_eye:)

Reply:
That makes sense. `of` and `mk` sound like constructors to me, but I like `as`.

Reply:
I'm marking this resolved, as it's been done.